tests/datavolume_test.go

```diff
@@ -119,6 +120,84 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		)
 	})
 
+	Describe("Verify DataVolume with block mode", func() {
+		var err error
+		var storageClass *storageV1.StorageClass
+		var pod *v1.Pod
+		var pv *v1.PersistentVolume
+
+		BeforeEach(func() {
+			By(fmt.Sprintf("Creating storageClass for Block PV"))
+			storageClass, err = f.CreateStorageClassFromDefinition(utils.NewStorageClassForBlockPVDefinition("manual"))
+			Expect(err).ToNot(HaveOccurred())
+
+			pod, err = utils.FindPodByPrefix(f.K8sClient, "cdi", "cdi-block-device", "kubevirt.io=cdi-block-device")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unable to get pod %q", "cdi"+"/"+"cdi-block-device"))
+
+			nodeName := pod.Spec.NodeName
+			By(fmt.Sprintf("Creating Block PV"))
+			pv, err = f.CreatePVFromDefinition(utils.NewBlockPVDefinition("local-volume1", "1Gi", nil, "manual", nodeName))
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify that PV's phase is Available.")
+			err = f.WaitTimeoutForPVReady(pv.Name, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := utils.DeletePV(f.K8sClient, pv)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = utils.DeleteStorageClass(f.K8sClient, storageClass)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		table.DescribeTable("should", func(name, command, url, dataVolumeName, eventReason string, phase cdiv1.DataVolumePhase) {
+			var dataVolume *cdiv1.DataVolume
+
+			switch name {
+			case "import-http":
+				dataVolume = utils.NewDataVolumeWithHTTPImportToBlockPV(dataVolumeName, "1Gi", url)
+			}
+			By(fmt.Sprintf("creating new datavolume %s", dataVolume.Name))
+			dataVolume, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("waiting for datavolume to match phase %s", string(phase)))
+			utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, phase, dataVolume.Name)
+			if err != nil {
+				PrintControllerLog(f)
+				dv, dverr := f.CdiClient.CdiV1alpha1().DataVolumes(f.Namespace.Name).Get(dataVolume.Name, metav1.GetOptions{})
+				if dverr != nil {
+					Fail(fmt.Sprintf("datavolume %s phase %s", dv.Name, dv.Status.Phase))
+				}
+			}
+			Expect(err).ToNot(HaveOccurred())
+
+			// verify PVC was created
+			By("verifying pvc was created")
+			_, err = f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(dataVolume.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprint("Verifying event occurred"))
+			Eventually(func() bool {
+				events, err := RunKubectlCommand(f, "get", "events", "-n", dataVolume.Namespace)
+				if err == nil {
+					fmt.Fprintf(GinkgoWriter, "%s", events)
+					return strings.Contains(events, eventReason)
+				}
+				fmt.Fprintf(GinkgoWriter, "ERROR: %s\n", err.Error())
+				return false
+			}, timeout, pollingInterval).Should(BeTrue())
+
```

Comment:
The test looks good, with one thing missing. Verifying that the content we imported actually ended up in the block device. Right now we have create the block DV, ran the importer pod, and checked the events that it completed, but we also have to verify that it did the right thing during the process.

To verify create a verifyer pod like we do with regular imports, and run an md5 checksum, or read the block device and do a byte by byte comparison with the source. Do something to make sure the contents of the block device match what we are expecting.

Reply:
I followed the block above (regular DV) where it does not check the content. Is that wrong?

Reply:
Yeah, that should verify the contents as well, not sure why it doesn't.

Reply:
opened an issue (https://github.com/kubevirt/containerized-data-importer/issues/722) to enhance Functional Tests for this feature. I need this patch to be merged ASAP. I will address the new issue a bit later.

Reply:
As a team we agreed that any new features (and this is definitely a new feature) should have appropriate unit and functional tests. I am okay with leaving the DV version of this test as is. **IF** the PVC version of this test does the actual verification of the content. Then we can address the missing verification of the DV tests later.